rest/blip_api_attachment_test.go

```diff
@@ -312,52 +312,60 @@ func TestBlipPushPullNewAttachmentCommonAncestor(t *testing.T) {
 
 		// CBL creates revisions 1-abc,2-abc on the client, with an attachment associated with rev 2.
 		bodyText := `{"greetings":[{"hi":"alice"}],"_attachments":{"hello.txt":{"data":"aGVsbG8gd29ybGQ="}}}`
-		err = btcRunner.StoreRevOnClient(btc.id, docID, "2-abc", []byte(bodyText))
+		rev := "2-abc"
+		if btc.UseHLV() {
+			rev = db.EncodeTestVersion("2@abc")
+		}
+		err = btcRunner.StoreRevOnClient(btc.id, docID, rev, []byte(bodyText))
 		require.NoError(t, err)
 
 		bodyText = `{"greetings":[{"hi":"alice"}],"_attachments":{"hello.txt":{"revpos":2,"length":11,"stub":true,"digest":"sha1-Kq5sNclPz7QV2+lfQIuc6R7oRu0="}}}`
 		revId, err := btcRunner.PushRevWithHistory(btc.id, docID, "", []byte(bodyText), 2, 0)
 		require.NoError(t, err)
-		assert.Equal(t, "2-abc", revId)
+		assert.Equal(t, rev, revId)
 
 		// Wait for the documents to be replicated at SG
 		_, ok := btc.pushReplication.WaitForMessage(2)
 		assert.True(t, ok)
 
-		resp := btc.rt.SendAdminRequest(http.MethodGet, "/{{.keyspace}}/"+docID+"?rev="+revId, "")
-		assert.Equal(t, http.StatusOK, resp.Code)
+		collection := rt.GetSingleTestDatabaseCollection()
+		doc, _, err := collection.GetDocWithXattr(ctx, docID, db.DocUnmarshalNoHistory)
+		require.NoError(t, err)
```

Comment:
Do we need to limit this test to xattr only? Or should we switch this (and other usages) to `collection.GetDocument()` ?

```suggestion
		doc, err := collection.GetDocument(ctx, docID, db.DocUnmarshalNoHistory)
		require.NoError(t, err)
```

Reply:
Good catch. GetDocument is a reasonable change for these tests.  For the next milestone we might want to disable VersionVectorSubtest inside BlipTestClientRunner if xattrs=false (as that's not a supported config when we start writing the HLV to its own xattr), but we don't need that yet.